{"cell_type":{"8fea91e4":"code","f4d0be9d":"code","51c10d44":"code","abf50a25":"code","4574f545":"markdown","88831476":"markdown","90f68468":"markdown","78b911c1":"markdown","da9e2d19":"markdown"},"source":{"8fea91e4":"# Basic example on how to buld a keras sequential model and train a classifier\n#!pip install tensorflow\n#!pip install keras\n#!pip install numpy\n#!pip install scikit-learn\n\nfrom __future__ import absolute_import\nfrom __future__ import division\nfrom __future__ import print_function\n\nfrom sklearn import datasets, metrics\nfrom keras.layers import Dense\nfrom keras.models import Sequential\n#from keras.utils import to_categorical\nfrom tensorflow.keras.utils import to_categorical\n\nimport numpy as np\nfrom sklearn.model_selection import train_test_split\n","f4d0be9d":"# Network model\ndef build_model():\n\t# create model\n\tmodel = Sequential()\n\tmodel.add(Dense(16, input_dim=4, activation='relu'))\n\tmodel.add(Dense(16, activation='relu'))\n\tmodel.add(Dense(16, activation='relu'))\n\tmodel.add(Dense(16, activation='relu'))\n\tmodel.add(Dense(3, activation='softmax'))\n\t# Compile model\n\tmodel.compile(loss='categorical_crossentropy', optimizer='sgd', metrics=['accuracy'])\n  # print structure\n\tmodel.summary()\n\treturn model","51c10d44":"\n# load dataset and convert to onehot encoding\niris = datasets.load_iris()\nx = iris.data\ny = to_categorical(iris.target)\n# print(y)\n# split dataset in train \/ test\nx_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.33, random_state=42)\n\n# build and train model\nclassifier = build_model()\nclassifier.fit(x_train, y_train, epochs=500)","abf50a25":"# evaluate model on test data\n_, score = classifier.evaluate(x_test,y_test)\n\nprint(\"Accuracy on test data: %f\" % score)","4574f545":"# 4. Score das \u00c9pocas e Acur\u00e1cia","88831476":"Do c\u00f3digo original, a \u00fanica mudan\u00e7a realizada foi a altera\u00e7\u00e3o do caminho da importa\u00e7\u00e3o para acessar a fun\u00e7\u00e3o *to_categorical*. Importando esta fun\u00e7\u00e3o da seguinte forma \"from keras.utils import to_categorical\" resultava em erro, pois ap\u00f3s a execu\u00e7\u00e3o do c\u00f3digo era retornado que essa fun\u00e7\u00e3o n\u00e3o existia no diret\u00f3rio informado. A solu\u00e7\u00e3o foi alterar a refer\u00eancia a mesma, acessando antes, a biblioteca do **tensorflow**.\n\nSolu\u00e7\u00e3o retirada do seguinte link: https:\/\/stackoverflow.com\/questions\/67018079\/error-in-from-keras-utils-import-to-categorical","90f68468":"# 2. Defini\u00e7\u00e3o do modelo de Rede","78b911c1":"# 3. Cria\u00e7\u00e3o do Pipeline de Classifica\u00e7\u00e3o","da9e2d19":"# 1. Importa\u00e7\u00e3o dos pacotes necess\u00e1rios"}}